src/tool_operate.c

```diff
@@ -751,10 +765,9 @@ static char *ipfs_gateway(void)
   if(gateway_file) {
     char *buf = NULL;
 
-    if((PARAM_OK == file2string(&buf, gateway_file)) && buf && *buf) {
-      bool add_slash = (buf[strlen(buf) - 1] != '/');
-      gateway = aprintf("%s%s", buf, (add_slash) ? "/" : "");
-    }
+    if((PARAM_OK == file2string(&buf, gateway_file)) && buf && *buf)
+      gateway = aprintf("%s", buf);
```

Comment:
seems like a useless copy?

Reply:
That's not entirely the case...

It gets the file content and checks it for non-null. If that is non-null than `gateway` is filled with it's value.
If the file would exist but is empty, this line would make `gateway` a non-null but empty variable.

That being said, i will restructure the checks to after the call which would clean it up nicely too.

Reply:
No, but copying `buf` to a new `gateway` buffer with `aprintf` is definitely a useless copy.